lib/purge-users-using-local-file.rb

```diff
@@ -0,0 +1,79 @@
+require 'zendesk_api'
+require 'rest-client'
+require 'json'
+require 'date'
+
+@client = ZendeskAPI::Client.new do |config|
+  config.url = ENV['ZENDESK_URL']
+  config.username = ENV['ZENDESK_USER_EMAIL']
+  config.password = ENV['ZENDESK_USER_PASSWORD']
+  config.retry = true
+end
+
+# Setup vars
+
+diag = "true"
+url = "#{ENV['ZENDESK_URL']}/deleted_users/"
+day = Date.today.next_day
+lastyear = day - 365
+
+input_file = File.open "data/soft_deleted_users.json"
+
+file_data = JSON.load input_file
+
+# Main loop
+file_data.each do |user|
+
+  # Extract some fields
+  updated_at = user["updated_at"]
+  last_login_at = user["last_login_at"]
+  user_id = user["id"]
+  active = user["active"]
+  name = user["name"]
+  # permanently_deleted = user["permanently_deleted"]
+
+  # Print some vars so we can corellate script actions & logic
+  if diag == "true"
+    # puts "user: #{user.inspect}"
+    puts "user_id: #{user_id}"
+    puts "name: #{name}"
+    puts "Active: #{active}"
+    puts "url: #{url}"
+    puts "day: #{day}"
+    puts "lastyear: #{lastyear}"
+    puts "last_login_at: #{last_login_at}"
+    puts "updated_at: #{updated_at}"
+    # puts "permanently_deleted: #{permanently_deleted}"
+    puts ". . . . . . . . . ."
+  end
+
+  # If permanently deleted, ignore this user_id
+  if name != "Permanently deleted user"
+    # Not permanently deleted so let's permanently delete
+    if active != "true"
+      puts "User #{user_id} already soft deleted, hard deleting..."
+      begin
+        # api does not support hard delete yet, so hard delete like this...
+          full_url = "#{url}#{user_id}.json"
+          RestClient::Request.execute method: :delete, url: full_url, user: ENV['ZENDESK_USER_EMAIL'], password: ENV['ZENDESK_USER_PASSWORD']
+    
+          rescue RestClient::UnprocessableEntity => api_error
+            puts "Received HTTP 422 from ZenDesk API for user #{user_id} => #{api_error}"
```

Comment:
It's unclear to me why we're catching 422 errors specifically, other than to print a different error to the screen. Surely that same information is included in the backtrace, which we print out in either case?

Reply:
@blairboy362  This came from Toby, it may be the case that we captured 422 first as an example, then following further errors, I added the 2nd block. It may be that if I now remove the 422 block, the script would function. However I'm not sure how I could setup a test to prove this due to it being a return from the API and I can't guarantee this :-( Where shall I go from here, it works and I don't believe it deserves maybe several hours work in order to test we can probably remove a couple of lines. Interested in your response!